lib/queue.js

```diff
@@ -30,16 +30,17 @@ const queue = (options) => {
         if ( ready ){
             const opts = Extend({}, DEFAULT_CONSUME_OPTIONS, consumeOptions);
             channel.consume(emitter.name, onMessage, opts, onConsume);
+            return;
         }
-        else {
-            emitter.once('ready', () => {
 
-                const opts = Extend({}, DEFAULT_CONSUME_OPTIONS, consumeOptions);
-                channel.consume(emitter.name, onMessage, opts, onConsume);
-            });
-        }
 
-        const onMessage = (msg) => {
+        emitter.once('ready', () => {
+
+            const opts = Extend({}, DEFAULT_CONSUME_OPTIONS, consumeOptions);
+            channel.consume(emitter.name, onMessage, opts, onConsume);
+        });
+
+        function onMessage(msg){
```

Comment:
You can appease the linter by [declaring a top-level `internals`][0] var, eg:

```
const internals = {};
```

and then you can attach this headless func to it, eg: 

```suggestion
        internals.onMessage: (msg) => {
```

And that should keep the linter happy :)

[0]: https://gist.github.com/hueniverse/a06f6315ea736ed1b46d#module-classes